server/src/main/java/org/opensearch/index/store/RemoteSegmentStoreDirectoryFactory.java

```diff
@@ -47,7 +48,7 @@ public Directory newDirectory(String repositoryName, IndexSettings indexSettings
             RemoteDirectory dataDirectory = createRemoteDirectory(repository, commonBlobPath, "data");
             RemoteDirectory metadataDirectory = createRemoteDirectory(repository, commonBlobPath, "metadata");
 
-            return new RemoteSegmentStoreDirectory(dataDirectory, metadataDirectory);
+            return new RemoteSegmentStoreDirectory(dataDirectory, metadataDirectory, RemoteSegmentMetadata.createMetadataParser());
```

Comment:
We can have this as a static constant in `RemoteSegmentStoreDirectory` as well, right?

Reply:
Yes, updated!